lib/spack/spack/fetch_strategy.py

```diff
@@ -319,6 +319,15 @@ def fetch(self):
             raise FailedDownloadError(url)
 
     def _existing_url(self, url):
+        # first, try the range request, if that fails
+        # try again with header request
+        if not self._check_existing_url_by_range_request(url):
```

Comment:
May I suggest reversing the condition for readability? Or even replacing the entire function with
```python
return self._check_existing_url_by_range_request(url) or self._check_existing_url_by_header_request(url)
```

Python evaluates `or` lazily (i.e. it won't evaluate the second argument if the first one is `True`)

Reply:
https://docs.python.org/3.8/reference/expressions.html#boolean-operations

Reply:
:+1: